### ISSUE TITLE
Type inference failed for Try.of()

### ISSUE THREAD

**Issue by hastebrot** [vavr-io/vavr-kotlin#6]
After an update to Kotlin 1.1.3 I receive the following compilation error for `Controls.kt`:

~~~
Error:(104, 25) Kotlin: Type mismatch: inferred type is () -> A
but CheckedFunction0<out A!>! was expected
~~~

A quick fix is to replace

~~~kt
fun <A> `try`(f: () -> A):
        Try<A> = Try.of(f)
~~~

with

~~~kt
fun <A> `try`(f: () -> A):
        Try<A> = Try.of<A>(object : CheckedFunction0<A> {
            override fun apply(): A = f()
        })
~~~

I wonder if there is a better way to fix this compilation error.

**Comment by hastebrot** [comment]
It's also possible to use `Try.ofSupplier()` in Vavr 0.9.

~~~kt
fun <A> `try`(f: () -> A):
        Try<A> = Try.ofSupplier(f)
~~~

**Comment by ruslansennov** [comment]
fixed 3892e200de9b4367ba76f42189e78f792fa8864f
